src/test/ui/issues/issue-61882.rs

```diff
@@ -0,0 +1,9 @@
+struct A<T>(T);
+
+impl A<bool> {
+    const B: A<u8> = Self(0);
+    //~^ ERROR mismatched types
+    //~| ERROR mismatched types
```

Comment:
Would be good to add the `const B: Self = Self(0);` case also for some redundancy?

I don't know what an interesting test for `A<&'static T>` would be... I have:

```rust
impl A<&'static u8> {
    fn f() -> Self {
        let x = 0;
        Self(&x)
    }
}
```

it fails with "cannot return value referencing local variable `x`".

Reply:
You can't return it. Try `Self(&x);` - that should still error, even if you're not returning it.

Reply:
Ah good call; it doesn't error:
```rust
struct A<T>(T);

impl A<&'static u8> {
    fn f() {
        let x = 0;
        Self(&x);
    }
}
```
so you can add that to the test file.

Reply:
Decided to check your branch out to help speed things up... but `Self(&x);` for `impl A<&'static u8> {` still does not error.

Reply:
It does error, but it needs to be in a separate test because it doesn't borrow-check `A::f` if `A::B` doesn't type-check (we should probably remove this limitation and keep going? cc @estebank)